lms/services/canvas.py

```diff
@@ -56,6 +56,27 @@ def can_see_file_in_course(self, file_id: str, course_id: str) -> bool:
 
         return False
 
+    def find_matching_file_in_course(self, course_id: str, file_: "models.File") -> str:
```

Comment:
Is `"models.File"` the right way to do the type annotation here?

Reply:
Not sure, I think you are expected to import `File` and use it here.

I only see this reference to using strings for thing that would be evaluable once the module is loaded: https://www.python.org/dev/peps/pep-0484/#forward-references

This is also possible:

```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from models import File
```

but ugly, if a module uses a type I think it's fine to import it.

Reply:
You could also take `file_name` and `file_size` instead of the full `File` but I don't think the importing or not importing the type should drive that decision.

Reply:
The function can return None so something like:

```suggestion
    def find_matching_file_in_course(self, course_id: str, file_: "models.File") -> Optional[str]:
```
and from typing import Optional at the top.

Reply:
Done, it's now `def find_matching_file_in_course(self, course_id: str, file_: models.File) -> Optional[str]:`